dataloader.py

```diff
@@ -0,0 +1,103 @@
+# PyTorch imports
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torch.utils.data.sampler import SubsetRandomSampler
+
+# Other libraries for data manipulation and visualization
+import os
+import numpy as np
+
+
+class PadSequence:
+    def __call__(self, batch):
+        #Each element in batch is (features, labels)
+        sorted_batch = sorted(batch, key=lambda x: x[0].shape[0], reverse=True)
+
+        sequences = [x[0] for x in sorted_batch]
+        sequences_padded = torch.nn.utils.rnn.pad_sequence(sequences, batch_first=True)
+
+        labels = [x[1] for x in sorted_batch]
+        labels_padded = torch.nn.utils.rnn.pad_sequence(labels, batch_first=True)
+        return sequences_padded, labels_padded
```

Comment:
The most important thing about working in an engineering team is making sure the code is transparent enough to be clear. I would recommend adding a docstring for this function and comments that clarify the purpose of each step with regards to the larger project, not just the features that are being manipulated.

Reply:
Yes, definitely I will add doc string and more comments to clarify codes.